Sprint 1/step 2

Nikba-Creative-Studio opened the pull request:
Проект. Написать весь JSX и сверстать

Nikba-Creative-Studio committed:
Подключение к API

1. Для выполнения запросов используется Fetch.
2. URL-адрес домена вынесен в отдельную константу.
3. Если при выполнении запроса к API произошли ошибки, они обрабатываются

Nikba-Creative-Studio committed:
Компоненты главной страницы: Modal и ModalOverlay

1. Modal — компонент самого модального окна: шапка с заголовком и иконка закрытия.
2. ModalOverlay — фоновая подложка под модальным окном.

Открытие модального окна:

1. Модальное окно открывается по клику на соответствующий элемент страницы.
2. Клик по ингредиенту открывает модальное окно с описанием ингредиента.
3. Клик по кнопке «Оформить заказ» открывает модальное окно с описанием заказа.

Закрытие модального окна:

1. Клик по иконке крестика закрывает модальное окно.
2. Клик по ModalOverlay или нажатие на клавишу “Esc” закрывают модальное окно.

Nikba-Creative-Studio committed:
Компоненты IngredientDetails и OrderDetails

1. Рефакторинг запуска модальных окон
2. IngredientDetails - содержит данные, полученные от API
3. OrderDetails - содержит фэйковые данные
4. Типизация - проверка типов PropTypes

Nikba-Creative-Studio committed:
Стилизация компонентов IngredientDetails и OrderDetails

irinamozes commented:
Проект очень хороший.

Взаимодействие с сервером происходит правильно. Окна показываются в портале.

Используется хук useCallback для предотвращения лишних переопределений одной и той же функции в функциональной компоненте.


Что нужно исправить.

1. Немного не по постановке задачи в описании задания выполнен показ окон в портале.
На всякий случай напоминаю, какая постановка задачи подразумевается при работе с порталом: из описания 2-го этапа 1-го спринта следует, что в проекте должно быть одно модальное окно (Modal, в котором и надо организовать портал), которое может наполняться каким угодно содержимым. На данном спринте это содержимое должно олицетворяться компонентами IngredientDetails и OrderDetails. То есть IngredientDetails и OrderDetails должны
показываться как children в Modal (в портале). В самих компонентах IngredientDetails и OrderDetails компонента Modal вызываться не должна.
ModalOverlay является элементом модального окна Modal (подложкой) и рендерится в Modal всегда. ModalOverlay рендерится как соседний элемент к children (это Вы делаете правильно).
Таким образом, сама компонента Modal должна вызываться в компонентах BurgerIngredients и BurgerConstructor для передачи ей IngredientDetails и OrderDetails как children.
А ещё лучше, если Modal будет вызываться два раза в App.

Nikba-Creative-Studio committed:
Рефакторинг

Modal вызывается в компонентах BurgerIngredients и BurgerConstructor для передачи ей IngredientDetails и OrderDetails как children.

irinamozes commented:
Проект отличный!

Что надо лучше.

1. Надо, чтобы слои с замками в верху и в низу собранного бургера, были одинаковыми булками.


Стандартный процесс работы реакта, если в нём не используется Redux, такой: переменные состояния определяются в родительском компоненте самого высокого уровня (App) в хуке useState, например, передаются в пропсах-коллбэках (определённых также в родительском компоненте) из родительского компонента в дочерние, меняют своё значение вследствие происхождения событий в дочерних компонентах. Эти изменённые переменные читаются в родительской и передаются из родительской компоненты другим дочерним компонентам, или передаются тем же самым дочерним компонентам, где они и были изменены. Таким образом, всё состояние проекта может находиться под контролем одной компоненты App, которая будет "единственным источником правды" для всего приложения.
Также этот паттерн применяется потому, что пропсы в компонентах менять нельзя. Их можно изменить только в той компоненте, где они определены, как переменные состояния.
А, вот как об этом же сказано здесь https://ru.react.js.org/docs/lifting-state-up.html:
"В React совместное использование состояния достигается перемещением его до ближайшего предка компонентов, которым оно требуется. Это называется «подъём состояния»."


Первый спринт принимается.

Желаю дальнейших успехов в учёбе и удачи!
